src/parser/position.rs

```diff
@@ -0,0 +1,77 @@
+use crate::parser::Span;
+#[cfg(not(target_arch = "wasm32"))]
+use proptest::prelude::*;
+
+/// Source code position of an expression in a file
+#[derive(Clone, Copy, Debug)]
+pub enum Pos {
+    No,
+    Pos {
+        from_offset: usize,
+        from_line: usize,
+        from_column: usize,
+        upto_offset: usize,
+        upto_line: usize,
+        upto_column: usize,
+    },
+}
+
+impl PartialEq for Pos {
+    fn eq(&self, _other: &Self) -> bool {
+        true
```

Comment:
Ahem, I’ve either missed something, or this deserves a spectacular comment.

Reply:
// This is so we can easily use derive(PartialEq) on datatypes like `Syntax` which contain `Pos`,
// since the source position an AST node comes from doesn't effect its equality